Sources/MongoSwift/ReadPreference.swift

```diff
@@ -134,36 +255,35 @@ public final class ReadPreference {
                         " \(MONGOC_SMALLEST_MAX_STALENESS_SECONDS), \(maxStalenessSeconds) given"
                 )
             }
-
-            mongoc_read_prefs_set_max_staleness_seconds(self._readPreference, maxStalenessSeconds)
+            mongoc_read_prefs_set_max_staleness_seconds(self.readPref, Int64(maxStalenessSeconds))
         }
     }
 
-    /// Initializes a new `ReadPreference` by copying an existing `ReadPreference`.
-    public init(from readPreference: ReadPreference) {
-        self._readPreference = mongoc_read_prefs_copy(readPreference._readPreference)
+    fileprivate var mode: ReadPreference.Mode {
+        ReadPreference.Mode(mongocMode: mongoc_read_prefs_get_mode(self.readPref))
     }
 
-    /// Initializes a new `ReadPreference` by copying a `mongoc_read_prefs_t`.
-    /// The caller is responsible for freeing the original `mongoc_read_prefs_t`.
-    internal init(from readPreference: OpaquePointer?) {
-        self._readPreference = mongoc_read_prefs_copy(readPreference)
-    }
+    fileprivate var tagSets: [Document]? {
+        guard let bson = mongoc_read_prefs_get_tags(self.readPref) else {
+            fatalError("Failed to retrieve read preference tags")
+        }
+        // we have to copy because libmongoc owns the pointer.
+        let wrapped = Document(copying: bson)
 
-    /// Cleans up internal state.
-    deinit {
-        guard let readPreference = self._readPreference else {
-            return
+        guard !wrapped.isEmpty else {
+            return nil
         }
-        mongoc_read_prefs_destroy(readPreference)
-        self._readPreference = nil
+
+        // swiftlint:disable:next force_unwrapping
+        return wrapped.values.map { $0.documentValue! } // libmongoc will always give us an array of documents
     }
-}
 
-/// An extension of `ReadPreference` to make it `Equatable`.
-extension ReadPreference: Equatable {
-    public static func == (lhs: ReadPreference, rhs: ReadPreference) -> Bool {
-        lhs.mode == rhs.mode && lhs.tagSets == rhs.tagSets &&
-            lhs.maxStalenessSeconds == rhs.maxStalenessSeconds
+    fileprivate var maxStalenessSeconds: Int? {
+        let maxStaleness = mongoc_read_prefs_get_max_staleness_seconds(self.readPref)
+        return maxStaleness == MONGOC_NO_MAX_STALENESS ? nil : Int(exactly: maxStaleness)
```

Comment:
this could technically return nil if you were on a 32 bit machine and somehow got a hold of a libmongoc read preference with a max staleness value greater than Int32.max, but that seems pretty unlikely. but if you think it would be better to fatal error or something here let me know.